### PR TITLE
Add extension to disable logging

### PR DESCRIPTION
**Pull request by lennykano**
Enables globally disabling Hugo via build.gradle, which is much easier than going through and deleting a whole bunch of annotations that I will want again some day.

### CODE REVIEW

**Comment by lennykano** [comment]
Fixes #87

File: hugo-runtime/src/main/java/hugo/weaving/internal/Hugo.java [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 @Aspect
 public class Hugo {
+
+  private static boolean sEnabled;
```
Comment (JakeWharton) [inline hugo-runtime/src/main/java/hugo/weaving/internal/Hugo.java]: No 's' prefix! Also you'll want this to be `volatile` or threads might not see value changes.

File: hugo-plugin/src/main/groovy/hugo/weaving/plugin/HugoExtension.groovy [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+    def logging(boolean enabled) {
+        logging = enabled
+    }
+
+    def logging() {
+        return logging;
+    }
+
+}
```
Comment (JakeWharton) [inline hugo-plugin/src/main/groovy/hugo/weaving/plugin/HugoExtension.groovy]: Add trailing newline

File: hugo-plugin/src/main/groovy/hugo/weaving/plugin/HugoExtension.groovy [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package hugo.weaving.plugin
+
+class HugoExtension {
+
+    def logging = true
```
Comment (JakeWharton) [inline hugo-plugin/src/main/groovy/hugo/weaving/plugin/HugoExtension.groovy]: this file should be two-space indent

File: hugo-plugin/src/main/groovy/hugo/weaving/plugin/HugoExtension.groovy [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    def logging(boolean enabled) {
```
Comment (JakeWharton) [inline hugo-plugin/src/main/groovy/hugo/weaving/plugin/HugoExtension.groovy]: This should be `setLogging` and the other should be `getLogging`, by convention

File: hugo-runtime/src/main/java/hugo/weaving/internal/Hugo.java [hunk 5]
```diff
@@ -31,6 +34,10 @@ public void method() {}
   @Pointcut("execution(@hugo.weaving.DebugLog *.new(..)) || constructorInsideAnnotatedType()")
   public void constructor() {}
 
+  public static synchronized void setEnabled(boolean enabled) {
```
Comment (JakeWharton) [inline hugo-runtime/src/main/java/hugo/weaving/internal/Hugo.java]: Doesn't need to be synchronized

**Comment by lennykano** [comment]
Thanks for looking at it, let me know if there's anything else you'd like changed.

**Comment by JakeWharton** [comment]
Squashed, rebased, and merged manually. Thanks!

**Comment by cesards** [comment]
@lennykano some documentation about new features in the README would be awesome :)

**Pull request closed without merging**